connectors/config.py

```diff
@@ -86,7 +86,7 @@ def _default_config():
             "initial_backoff_duration": 1,
             "backoff_multiplier": 2,
             "log_level": "info",
-            "feature_use_connectors_api": False,
+            "feature_use_connectors_api": True,
```

Comment:
This should be reflected in the PR - the main goal of the PR is to toggle this flag, tests are side-effect

Reply:
If we're updating the flag in this PR, we should also update the example config files. They default to false.

https://github.com/elastic/connectors/blob/7d82cf09dd9e463a48d028871c6e8dcca86680f6/config.yml.example#L156

Also I'll close https://github.com/elastic/connectors/pull/2884 in favor of this PR.

Reply:
> If we're updating the flag in this PR, we should also update the example config files. They default to false.

Ahh yes! I will update this in a followup PR, I missed your comment and don't want to re-trigger the CI (all ftest) to include the proposed change in this PR